src/ansible_compat/prerun.py

```diff
@@ -0,0 +1,427 @@
+    was detected.
+    """
+
+    @lru_cache()
+    def _get_ver_err() -> Tuple[str, str]:
+
+        err = ""
+        failed = False
+        ver = ""
+        result = subprocess.run(
+            args=["ansible", "--version"],
+            stdout=subprocess.PIPE,
+            universal_newlines=True,
+            check=False,
+        )
+        if result.returncode != 0:
+            return (
+                ver,
+                "FATAL: Unable to retrieve ansible cli version: %s" % result.stdout,
+            )
+
+        ver, error = parse_ansible_version(result.stdout)
+        if error is not None:
+            return "", error
+        try:
+            # pylint: disable=import-outside-toplevel
+            from ansible.release import (
+                __version__ as ansible_module_version,  # type: ignore
+            )
+
+            if version.parse(ansible_module_version) < version.parse(
+                ANSIBLE_MIN_VERSION
+            ):
+                failed = True
+        except (ImportError, ModuleNotFoundError) as e:
+            failed = True
+            ansible_module_version = "none"
+            err += f"{e}\n"
+        if failed:
+            err += (
+                "FATAL: ansible-lint requires a version of Ansible package"
+                " >= %s, but %s was found. "
+                "Please install a compatible version using the same python interpreter. See "
+                "https://docs.ansible.com/ansible/latest/installation_guide"
+                "/intro_installation.html#installing-ansible-with-pip"
+                % (ANSIBLE_MIN_VERSION, ansible_module_version)
+            )
+
+        elif ver != ansible_module_version:
+            err = (
+                f"FATAL: Ansible CLI ({ver}) and python module"
+                f" ({ansible_module_version}) versions do not match. This "
+                "indicates a broken execution environment."
+            )
+        return ver, err
+
+    ver, err = _get_ver_err()
+    if exit_on_error and err:
+        _logger.error(err)
+        sys.exit(ANSIBLE_MISSING_RC)
+    return ver, err
+
+
+def install_collection(collection: str, destination: Optional[str] = None) -> None:
+    """Install an Ansible collection.
+
+    Can accept version constraints like 'foo.bar:>=1.2.3'
+    """
+    cmd = [
+        "ansible-galaxy",
+        "collection",
+        "install",
+        "-v",
+    ]
+    if destination:
+        cmd.extend(["-p", destination])
+    cmd.append(f"{collection}")
+
+    _logger.info("Running %s", " ".join(cmd))
+    run = subprocess.run(
+        cmd,
+        universal_newlines=True,
+        check=False,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+    )
+    if run.returncode != 0:
+        _logger.error("Command returned %s code:\n%s", run.returncode, run.stdout)
+        sys.exit(INVALID_PREREQUISITES_RC)
+
+
+@tenacity.retry(  # Retry up to 3 times as galaxy server can return errors
```

Comment:
Why was this lib chosen? My usual go-to choice is `backoff`. Is it somehow better?

Reply:
Also, the next version of ansible-core should come with its own retry logic so maybe it's worth making it conditional.

Reply:
Added to https://github.com/ansible-community/ansible-compat/issues/6